pySDC/core/Sweeper.py

```diff
@@ -270,40 +280,96 @@ def rho(x):
             self.parallelizable = True
 
         elif qd_type == "MIN-SR-S":
+
```

Comment:
There is quite a lot of code in this `if`-`else` chain. I would advocate for splitting this up into functions and only calling them here.

Reply:
Indeed, but this will come with an other PR later, including a simplification of the QDelta matrix generation ...